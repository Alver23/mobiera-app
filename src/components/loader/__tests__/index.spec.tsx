// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ActivityIndicator } from 'react-native';

// Components
import StatusBar from '@mobiera/components/status-bar';

// Under test file
import Loader from '../index';

describe('<Loader />', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Loader />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(component.find(ActivityIndicator).exists()).toBeTruthy();
    expect(component.find(StatusBar).exists()).toBeTruthy();
  });
});
