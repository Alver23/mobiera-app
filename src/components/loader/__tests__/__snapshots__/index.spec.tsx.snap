// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should save a snapshot of the component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 3,
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#2089dc"
    hidesWhenStopped={true}
    size="large"
  />
  <StatusBarComponent />
</View>
`;
